pytest_html/resources/main.js

```diff
@@ -190,3 +190,33 @@ function one_row_for_data() {
         }
     });
 }
+
+function filter_table(elem) {
+    var outcome_att = "data-test-result";
+    var results_table = document.getElementById("results-table");
+
+    var rows = results_table.rows; 
+    var outcome = elem.getAttribute(outcome_att);
+    var hidden = !elem.checked;
+
+    //It hides all the rows that match the outcome
+    for(var i = 0; i < rows.length; i++){
+        if(rows[i].getAttribute(outcome_att) == outcome) {
+            rows[i].hidden = hidden;
+        }
+    }
+
+    hide_extra_html_by_attribute(hidden, outcome_att, outcome);
+}
+
+function hide_extra_html_by_attribute(hidden, attribute_key, attribute_value) {
```

Comment:
We just had a quick chat about this, but let's explore grouping two rows within a `<tbody>`, and then using this for toggling display of both rows. This save us from splitting the rows using JavaScript in the first place, and also means the report looks better when JavaScript is disabled. I uploaded a rough (mostly) working example here: http://people.mozilla.org/~dhunt/tbody-report.html